File: integrations/src/main/java/com/nivasafinance/services/voice/provider/exotel/data/ExotelConfiguration.java
package com.nivasafinance.services.voice.provider.exotel.data;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ExotelConfiguration {
    private String accountSid;
    private String baseUrl;
    private String callWebhookUrl;
    private String campaignWebhookUrl;
    private String campaignCallWebhookUrl;
    private String apiKey;
    private String apiToken;
}

File: integrations/src/main/kotlin/services/voice/provider/exotel/data/ExotelConfiguration.kt
package services.voice.provider.exotel.data

data class ExotelConfiguration(
    val accountSid: String,
    val authToken: String,
    val subdomain: String,
    val callerId: String,
    val webhookUrl: String,
    val apiKey: String = "",
    val apiToken: String = "",
    val recordingEnabled: Boolean = true,
    val maxCallDuration: Int = 3600,
    val retryAttempts: Int = 3,
    val timeout: Int = 30
)
